geoh5py/data/reference_value_map.py

```diff
@@ -19,19 +19,30 @@
 
 from abc import ABC
 
+import numpy as np
+
 
 class ReferenceValueMap(ABC):
     """Maps from reference index to reference value of ReferencedData."""
 
-    def __init__(self, color_map: dict[int, str] | None = None):
-        self._map = {} if color_map is None else color_map
+    def __init__(self, color_map: dict[int, str]):
+        self.map: dict[int, str] = color_map
 
     def __getitem__(self, item: int) -> str:
         return self._map[item]
 
     def __setitem__(self, key, value):
+        if not isinstance(key, (int, np.int16, np.int32, np.int64)) or key < 0:
+            raise KeyError("Key must be an positive integer")
+        if key == 0:
+            if value != "Unknown" and not (
+                len(self) <= 2 and value == "False" and value == "True"
+            ):
+                raise ValueError("Value for key 0 must be 'Unknown'")
+        if not isinstance(value, str):
+            raise TypeError("Value must be a string")
```

Comment:
Can we turn this into a small static method to reduce code duplication with the map.setter?